x/amm/types/math.go

```diff
@@ -34,43 +35,43 @@ func LiquidityForAmounts(currentSqrtPrice, sqrtPriceA, sqrtPriceB sdk.Dec, amt0,
 	return LiquidityForAmount1(sqrtPriceA, sqrtPriceB, amt1)
 }
 
-func Amount0DeltaRounding(sqrtPriceA, sqrtPriceB sdk.Dec, liquidity sdk.Int, roundUp bool) sdk.Int {
+func Amount0DeltaRounding(sqrtPriceA, sqrtPriceB cremath.BigDec, liquidity sdk.Int, roundUp bool) sdk.Int {
 	if sqrtPriceA.GT(sqrtPriceB) {
 		sqrtPriceA, sqrtPriceB = sqrtPriceB, sqrtPriceA
 	}
-	intermediate := sqrtPriceB.Sub(sqrtPriceA).MulInt(liquidity)
+	intermediate := sqrtPriceB.Sub(sqrtPriceA).MulIntMut(liquidity)
 	if roundUp {
-		return intermediate.QuoRoundUp(sqrtPriceB).QuoRoundUp(sqrtPriceA).Ceil().TruncateInt()
+		return intermediate.QuoRoundUpMut(sqrtPriceB).QuoRoundUpMut(sqrtPriceA).Ceil().TruncateInt()
```

Comment:
What about using `intermediate.QuoRoundUpMut(sqrtPriceB.Mul(sqrtPriceA)).Ceil().TruncateInt()` instead of `intermediate.QuoRoundUpMut(sqrtPriceB).QuoRoundUpMut(sqrtPriceA).Ceil().TruncateInt()` in order to avoid using twice of round up?